CONN-2914 Add New JFrog Cloud-To-Cloud Source (#3730)
SumoLogic/sumologic-documentation: docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/jfrog-xray.md

---
id: jfrog-xray-source
title: JFrog Xray Source
sidebar_label: JFrog Xray
tags:
  - cloud-to-cloud
  - jfrog
  - xray
description: Learn how to collect data from the JFrog Xray platform.
---
import CodeBlock from '@theme/CodeBlock';
import ExampleJSON from '/files/c2c/jfrog-xray/example.json';
import MyComponentSource from '!!raw-loader!/files/c2c/jfrog-xray/example.json';
import TerraformExample from '!!raw-loader!/files/c2c/jfrog-xray/example.tf';
import useBaseUrl from '@docusaurus/useBaseUrl';

<img src={useBaseUrl('img/integrations/security-threat-detection/jfrog.png')} alt="thumbnail icon" width="85"/>

> The JFrog Xray source will collect JFrog Xray violations by querying the API for new policy violations when they occur.

Reviewer: ```suggestion
The Sumo Logic JFrog Xray source collects JFrog Xray violations by querying the API for new policy violations when they occur.
```

:::note
This source is available in the [Fed deployment](/docs/api/getting-started#sumo-logic-endpoints-by-deployment-and-firewall-security).
:::

## Data collected

| Polling Interval | Data |
| :--- | :--- |
| 5 minutes |  [Get Violations Endpoint](https://jfrog.com/help/r/xray-rest-apis/get-violations)|

## Setup

### Vendor configuration

> Ensure Xray has policies configured. This source will ingest violation logs when those policies raise violations. You will need the domain of your JFrog instance and JFrog username/password in order to configure this source. We suggest using a dedicated user account for this source.

Reviewer: ```suggestion
Make sure that you've set up Xray's policies correctly. This source will capture violation logs triggered by those policies. To configure this source, you'll need the domain of your JFrog instance and your JFrog username/password. We recommend creating a dedicated user account specifically for this purpose.
```

### Source configuration

When you create a JFrog Xray Source, you add it to a Hosted Collector. Before creating the Source, identify the Hosted Collector you want to use or create a new Hosted Collector. For instructions, see [Configure a Hosted Collector](/docs/send-data/hosted-collectors/configure-hosted-collector).

To configure the JFrog Xray Source:
1. In Sumo Logic, select **Manage Data** > **Collection** > **Collection**. 
1. On the Collectors page, click **Add Source** next to a Hosted Collector.

> 1. Search for and select **JFrog Xray** icon.

Reviewer: ```suggestion
1. Search for and select the **JFrog Xray** icon.
```

> 1. Enter a **Name** to display for the Source in the Sumo Logic web application. The description is optional.

Reviewer: ```suggestion
1. Enter a **Name** to display for the Source in Sumo Logic. The description is optional.
```

1. (Optional) For **Source Category**, enter any string to tag the output collected from the Source. Category metadata is stored in a searchable field called `_sourceCategory`. 
1. (Optional) **Fields**. Click the **+Add Field** link to define the fields you want to associate. Each field needs a name (key) and value.
   * ![green check circle.png](/img/reuse/green-check-circle.png) A green circle with a check mark is shown when the field exists in the Fields table schema.
   * ![orange exclamation point.png](/img/reuse/orange-exclamation-point.png) An orange triangle with an exclamation point is shown when the field doesn't exist in the Fields table schema. In this case, an option to automatically add the nonexistent fields to the Fields table schema is provided. If a field is sent to Sumo Logic that does not exist in the Fields schema it is ignored, known as dropped.

> 1. In **JFrog Base URL**, enter your Jfrog instance domain. E.g. `https://acme.jfrog.io`

Reviewer: ```suggestion
1. In **JFrog Base URL**, enter your JFrog instance domain (for example, `https://acme.jfrog.io`).
```

> 1. In **HTTP Basic Auth Username**, enter your JFrog username you created
> 1. In **HTTP Basic Auth Password**, enter your JFrog password you created

Reviewer: ```suggestion
1. In **HTTP Basic Auth Username**, enter your JFrog username you created.
1. In **HTTP Basic Auth Password**, enter your JFrog password you created.
```

1. In **Collect Violation Details**, only toggle this on if you need the full violation details. This will require 1 extra API call per violation log.

> 1. (Optional) The **Polling Interval** is set for 5 minutes default, you can adjust it based on your needs.

Reviewer: ```suggestion
1. (Optional) The **Polling Interval** is set for 5 minutes by default. You can adjust it based on your needs.
```
